### PR TITLE
Mention bash in prerequisites

### PR DESCRIPTION
**Pull request by haampie**
Isn't installed on Alpine.

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]
@tgamblin should we make the compiler wrappers POSIX sh compliant?

File: lib/spack/docs/getting_started.rst [hunk 1]
```diff
@@ -17,7 +17,8 @@ Spack has the following minimum requirements, which must be installed
 before Spack is run:
 
 #. Python 2 (2.6 or 2.7) or 3 (3.5 - 3.9) to run Spack
-#. A C/C++ compiler for building
+#. A C/C++ compiler for building and the ``bash`` shell for Spack's compiler
+   wrapper
```
Comment (tgamblin) [inline lib/spack/docs/getting_started.rst:21]: Add a note here that you only need it for building; it doesn’t have to be your shell. For example, you can run spack in containers where the root shell is dash or ash. You just can’t build at the moment without bash installed.

**Comment by tgamblin** [comment]
> should we make the compiler wrappers POSIX sh compliant?

it’s tempting, and I think “yes” is probably the right answer eventually, but I think adding the note is good for now.

**Comment by tgamblin** [comment]
The real question is how maintainable they’d be without bash arrays. Right now we make pretty extensive use of arrays, and I don’t know if we can actually do exactly the same thing in dash without losing correctness for some corner cases. I *think* we can do it, but I am not 100% sure.

**Pull request merged**